src/transaction_builder.ts

```diff
@@ -0,0 +1,407 @@
+import {
+    AuthorizedKey,
+    AuthorizedKeyUserInput,
+    Curve,
+    HashAlgorithm,
+    TransactionData,
+    UserTypeTransaction
+} from "./types.js"
+import {
+    bigIntToUint8Array,
+    concatUint8Arrays,
+    intToUint8Array,
+    maybeHexToUint8Array,
+    maybeStringToUint8Array,
+    toByteArray,
+    uint8ArrayToHex
+} from "./utils.js";
+import {deriveAddress, deriveKeyPair, sign} from "./crypto.js";
+
+const VERSION = 1
+
+const txTypes = {
```

Comment:
Is this variable still relevant as you used an enum for the tx type ?

Reply:
I think yes because I need the tx type number and not the it's name

Reply:
ok so you can have multiple alternatives to still use type system:
- use another enum 
- replace enum by an object
- create a function which take an enum member and return its number

Reply:
I think best option is the function